src/Path/Include.hs

```diff
@@ -723,31 +728,48 @@ mkRelFile = either (error . show) lift . parseRelFile
 --------------------------------------------------------------------------------
 -- Internal functions
 
--- | Internal use for normalizing a directory.
+-- | Normalizes directory path with platform-specific rules.
 normalizeDir :: FilePath -> FilePath
 normalizeDir =
       normalizeRelDir
     . FilePath.addTrailingPathSeparator
     . normalizeFilePath
-  where
-      -- Represent a "." in relative dir path as "" internally so that it
-      -- composes without having to renormalize the path.
-      normalizeRelDir p | p == relRootFP = ""
-      normalizeRelDir p = p
-
+  where -- Represent a "." in relative dir path as "" internally so that it
+        -- composes without having to renormalize the path.
+        normalizeRelDir p
+          | p == relRootFP = ""
+          | otherwise = p
+
+-- | Replaces consecutive path seps with single sep and replaces alt sep with standard sep.
+normalizeAllSeps :: FilePath -> FilePath
+normalizeAllSeps = foldr normSeps []
+  where normSeps ch [] = [ch]
+        normSeps ch path@(p0:_)
+          | FilePath.isPathSeparator ch && FilePath.isPathSeparator p0 = path
+          | FilePath.isPathSeparator ch = FilePath.pathSeparator:path
+          | otherwise = ch:path
+
+-- | Normalizes seps except at the beginning of path.
+normalizeSepsExceptLeading :: FilePath -> FilePath
+normalizeSepsExceptLeading path = normLeadingSeps ++ normalizeAllSeps rest
+  where (leadingSeps, rest) = span FilePath.isPathSeparator path
+        normLeadingSeps = replicate (min 2 (length leadingSeps)) FilePath.pathSeparator
```

Comment:
Given that we in fact do some leading path sep normalization, I guess this function deserves a better name and more correct documentation. How about `normalizeSepsWin` for example? That would also make it clear that it's Windows-specific.

Reply:
Renamed to `normalizeWindowsSeps` and expanded comment.

I wanted those internal functions to be more reusable, but yeah that one is too specific.